src/components/entries/List.js

```diff
@@ -0,0 +1,208 @@
+import {
+  useEffect,
+  useRef,
+  useState
+} from 'preact/hooks';
+
+import {
+  useKeyFactory,
+  usePrevious
+} from '../../hooks';
+
+import classnames from 'classnames';
+
+import {
+  CreateIcon,
+  GroupArrowIcon,
+  ListDeleteIcon
+} from '../icons';
+
+/**
+ * Entry for handling lists represented as nested entries.
+ *
+ * @template Item
+ * @param {object} props
+ * @param {string} props.id
+ * @param {*} props.element
+ * @param {Function} props.onAdd
+ * @param {Function} props.renderItem
+ * @param {string} [props.label='<empty>']
+ * @param {Function} [props.onRemove]
+ * @param {Item[]} [props.items]
+ * @param {boolean} [props.open]
+ * @param {(a: Item, b: Item) => -1 | 0 | 1} [props.compareFn]
```

Comment:
Can we have any reasonable default here? Probably not, right? Because

* every item has a different structure (name, label, value, ...)
* the absence of the `compareFn` indicates disabling of the sorting

Reply:
I don't think it's possible. Opposed to ListGroup, we don't know the structure of `Item` upfront. Thus, it's impossible to predict a default property for sorting.

Reply:
That's fine 👍